### PR TITLE
23 Jan/2024 ✅ Getting Articles and Trendings by category is done!

### PR DESCRIPTION
**Pull request by asliddintursunov**
I finished getting trendings and articles by its category when category button is clicked.

### CODE REVIEW

**Comment by asliddintursunov** [comment]
Need to **Review** from **Jan 23** to **Jan 26** commits!

File: app/auth/components/Username.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        style={{
+          borderColor: RegExp === false ? "red" : "",
+        }}
+        placeholder="User0046"
+        onChange={(e) =>
+          setUserData((prev) => ({ ...prev, username: e.target.value }))
+        }
+        required
+      />
+      {RegExp === false && (
```
Comment (muslimchikkk) [inline app/auth/components/Username.tsx:23]: RegExp seems a bit confusing. Maybe we need to rename it to something like `isUserNameValid`?

File: app/auth/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"use client";
+type Props = {
+  authType: "signin" | "signup";
+};
+
+function Button({ authType }: Props) {
+  return (
+    <button className="btn btn-circle btn-warning w-full">
+      {authType === "signin" && "SIGN IN"}
```
Comment (muslimchikkk) [inline app/auth/components/Button.tsx:9]: How about making this Button even more generic by passing the button title as a prop? With this you can use that button everywhere not only in auth page/modal

File: app/auth/components/Email.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        }}
+        placeholder="example@gmail.com"
+        onChange={(e) =>
+          setUserData((prev) => ({ ...prev, email: e.target.value }))
+        }
+        required
+      />
+      {RegExp === false && (
+        <span className="text-xs text-red-500">
+          Double check your email, somethiong&#39;s missing!
```
Comment (muslimchikkk) [inline app/auth/components/Email.tsx:25]: Maybe we should place Uzbek text right away?

File: components/Navbar.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import Link from "next/link";
 import ScrollToTop from "@/components/ScrollToTop";
 export default function Navbar() {
-  const isRegistered = true;
+  const isRegistered = false;
```
Comment (muslimchikkk) [inline components/Navbar.tsx:4]: can we get this value from the URL params until we finish up registration?

**Review by muslimchikkk — APPROVED** [review]
Looks great, I think we can remove all the unnecessary console.logs across the code base. Wondering how and for what you are going to use Redux?

File: app/auth/components/Username.tsx (hunk 1 above)
Comment (asliddintursunov) [reply app/auth/components/Username.tsx:23]: Sure, I will change it.

File: app/auth/components/Button.tsx (hunk 2 above)
Comment (asliddintursunov) [reply app/auth/components/Button.tsx:9]: When I created the button component in the auth directory, I meant it to be used for only auth. If it's better to change it, I will.

File: components/Navbar.tsx (hunk 4 above)
Comment (asliddintursunov) [reply components/Navbar.tsx:4]: Should a backend be written for it or not?

**Pull request merged**